sway-core/src/asm_generation/expression/mod.rs

```diff
@@ -276,48 +257,36 @@ pub(crate) fn convert_expression_to_asm(
         TypedExpressionVariant::TupleElemAccess {
             resolved_type_of_parent,
             prefix,
-            elem_to_access_num,
             elem_to_access_span,
-        } => convert_subfield_expression_to_asm(
-            &exp.span,
-            prefix,
-            &format!("{}", elem_to_access_num),
-            elem_to_access_span.clone(),
-            *resolved_type_of_parent,
-            namespace,
-            register_sequencer,
-            return_register,
-        ),
-        /*
-        TypedExpressionVariant::EnumArgAccess {
-            prefix,
-            variant_to_access,
-            arg_num_to_access,
-            resolved_type_of_parent,
-        } => convert_enum_arg_expression_to_asm(
-            &exp.span,
-            prefix,
-            variant_to_access,
-            arg_num_to_access.to_owned(),
-            *resolved_type_of_parent,
-            namespace,
-            register_sequencer,
-            return_register,
-        ),
-        */
+            elem_to_access_num,
+        } => {
+            // sorry
+            let leaked_ix: &'static str = Box::leak(Box::new(elem_to_access_num.to_string()));
+            let access_ident = Ident::new_with_override(leaked_ix, elem_to_access_span.clone());
```

Comment:
Obviously less than ideal.  Should we make an issue to not need `'static` strings for `Ident`?

Reply:
I'm not sure. I was debating the performance costs. We clone `Ident` a _lot_ throughout the compiler. I somewhat suspect that this leak is better for performance than using `String` everywhere for `Ident`. And using a scoped lifetime, like `<'sc>`, would be what we had before and nobody liked that.

This only happens specifically in these tuple accesses. I'm sure we can sort out a more localized solution that just updates the tuple access stuff. Maybe that's what the issue should be about.